diff --git a/components/esp_event/linker.lf b/components/esp_event/linker.lf
index c10dc4c6d..97ac1702f 100644
--- a/components/esp_event/linker.lf
+++ b/components/esp_event/linker.lf
@@ -1,6 +1,8 @@
-if ESP_EVENT_POST_FROM_IRAM_ISR = y:
-    [mapping:esp_event]
-    archive: libesp_event.a
-    entries:
+[mapping:esp_event]
+archive: libesp_event.a
+entries:
+    if ESP_EVENT_POST_FROM_IRAM_ISR = y:
         esp_event:esp_event_isr_post_to (noflash)
         default_event_loop:esp_event_isr_post (noflash)
+    else:
+        * (default)
diff --git a/tools/cmake/build.cmake b/tools/cmake/build.cmake
index 4e29c8625..afdda1557 100644
--- a/tools/cmake/build.cmake
+++ b/tools/cmake/build.cmake
@@ -394,7 +394,7 @@ macro(idf_build_process target)
     __build_set_default(SDKCONFIG_DEFAULTS "")
 
     # Check for required Python modules
-    __build_check_python()
+    #__build_check_python()
 
     idf_build_set_property(__COMPONENT_REQUIRES_COMMON ${target} APPEND)
 
